library/std/src/sync/mutex.rs

```diff
@@ -219,6 +219,23 @@ impl<T> Mutex<T> {
             data: UnsafeCell::new(t),
         }
     }
+
+    /// Immediately drops the guard, and consequently unlocks the mutex.
+    ///
+    /// This is equivalent to calling [`Drop::drop`] on the guard, but is more self-documenting.
```

Comment:
Shouldn't this refer to `std::mem::drop` - one can't call Drop's method?

IMO the doc comment could be even more clear that letting the guard go out of scope or using `drop` is just fine, but this method would be available if wanted.